Setup CircleCI

We should do a simple CircleCI setup, at least to do linting. Side View is a good example: https://github.com/mozilla/side-view/tree/master/.circleci

For the _"'React' must be in scope when using JSX  [react/react-in-jsx-scope](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md)"_ errors, I'd just disable that rule in the [.eslintrc.js](https://github.com/mozilla/email-tabs/blob/master/.eslintrc.js) config file.

You'll still have a few of the "'React' is not defined" errors (see below), but those can be defined in the .eslintrc.js file as `globals`:

```js
  "globals": {
    emailTemplates: false,
    React: false,
    ReactDOMServer: false
  },
```

```
/Users/pdehaan/dev/github/mozilla/email-tabs/addon/emailTemplates.jsx
   5:23  error  'React' is not defined           no-undef
  17:26  error  'React' is not defined           no-undef
  55:21  error  'ReactDOMServer' is not defined  no-undef
```

I think after that (and running <kbd>$ npm run lint:js -- --fix</kbd>), you should only have about 7 errors left:

```sh
$ npm run lint:js -- --fix

> email-tabs@0.1.0 lint:js /Users/pdehaan/dev/github/mozilla/email-tabs
> eslint addon --ext=js,jsx "--fix"


/Users/pdehaan/dev/github/mozilla/email-tabs/addon/background.js
    4:27  error  Expected '===' and instead saw '=='                 eqeqeq
    9:27  error  Expected '===' and instead saw '=='                 eqeqeq
   88:3   error  Unsafe assignment to innerHTML                      no-unsanitized/property
  107:3   error  Function 'loginInterrupt' expected no return value  consistent-return
  113:3   error  Unreachable code                                    no-unreachable

/Users/pdehaan/dev/github/mozilla/email-tabs/addon/popup.jsx
  147:71  error  HTML entities must be escaped  react/no-unescaped-entities

/Users/pdehaan/dev/github/mozilla/email-tabs/addon/set-html-email.js
  113:3  error  Unsafe assignment to innerHTML  no-unsanitized/property

✖ 7 problems (7 errors, 0 warnings)
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,0 +1,34 @@
+version: 2.0
+jobs:
+    package:
+        docker:
+            - image: circleci/node:8.9.4
+        steps:
+            - checkout
+            - restore_cache:
+                keys:
+                    - node-modules-
+            - run:
+                name: install
+                command: npm install
+            - run:
+                name: package
+                command: npm run package
+            - run:
+                name: lint
+                command: npm run lint
+            - save_cache:
+                key: node-modules-
+                paths:
+                    - node_modules
+            - store_artifacts:
+                path: addon.xpi
+            - persist_to_workspace:
+                root: .
+                paths:
+                    - ./*
+workflows:
+    version: 2
+    package:
+        jobs:
+            - package

